src/background/stores/card-selection-store.ts

```diff
@@ -24,17 +24,72 @@ export class CardSelectionStore extends BaseStoreImpl<CardSelectionStoreData> {
 
     public getDefaultState(): CardSelectionStoreData {
         const defaultValue: CardSelectionStoreData = {
-            rules: null,
+            rules: {},
         };
 
         return defaultValue;
     }
 
-    public toggleRuleExpandCollapse(payload: RuleExpandCollapsePayload): void {}
+    private deselectAllCardsInRule = (rule: RuleExpandCollapseData): void => {
+        if (!rule) {
+            return;
+        }
 
-    public toggleCardSelection(payload: CardSelectionPayload): void {}
+        for (const resultInstanceUid of Object.keys(rule.cards)) {
+            rule.cards[resultInstanceUid] = false;
+        }
+    };
 
-    public CollapseAllRules(): void {}
+    public toggleRuleExpandCollapse = (payload: RuleExpandCollapsePayload): void => {
```

Comment:
I don't believe this needs to be a public method.

Reply:
Done.